.github/actions/python-poetry-install/action.yml

```diff
@@ -0,0 +1,26 @@
+name: "Python & Poetry install and cache"
+description: "Python & Poetry install and cache"
+
+inputs:
+  poetry-version:
+    description: "Poetry version"
+    required: true
+  python-version:
+    description: "Python version"
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+    - name: Install Poetry
+      env:
+        POETRY_VERSION: ${{ inputs.poetry-version }}
+      shell: bash
+      run: |
+        curl -sSL https://install.python-poetry.org | python -
+        echo "$HOME/.poetry/bin" >> $GITHUB_PATH
+
+    - uses: actions/setup-python@v4
+      with:
+        python-version: ${{ inputs.python-version }}
+        cache: 'poetry'
```

Comment:
I enable the `cache: 'poetry'` option following https://github.com/actions/setup-python#caching-packages-dependencies 👀

Reply:
Note the following comment on https://github.com/QunaSys/quri-parts/pull/116:

> - In each workflow (`lint`, `typecheck`, `test` and `doc`), only necessary dependencies are installed with `poetry install --only [groups]`
>   - Since installed dependecies change depending on the workflow, I disabled caching of poetry virtualenv. (It may be safe to keep it enabled, but it is not likely that the caching provides much improvement on the execution time)